packages/wonder-blocks-data/util/response-cache.test.js

```diff
@@ -0,0 +1,346 @@
+// @flow
+import {ResponseCache} from "./response-cache.js";
+
+import type {IRequestHandler} from "./types.js";
+
+describe("./response-cache.js", () => {
+    afterEach(() => {
+        /**
+         * This is needed or the JSON.stringify mocks need to be
+         * mockImplementationOnce. This is because if the snapshots need
+         * to update, they write the inline snapshot and that appears to invoke
+         * prettier which in turn, calls JSON.stringify. And if that mock
+         * throws, then boom. No snapshot update and a big old confusing test
+         * failure.
+         */
+        jest.restoreAllMocks();
+    });
+
+    describe("#initialize", () => {
+        it("should initialize the cache with the given data", () => {
+            // Arrange
+            const cache = new ResponseCache();
+            const fakeHandler: IRequestHandler<string, string> = {
+                getKey: () => "MY_KEY",
+                type: "MY_HANDLER",
+                invalidateCache: () => false,
+                fulfillRequest: jest.fn(),
+            };
+
+            // Act
+            cache.initialize({
+                MY_HANDLER: {
+                    MY_KEY: {data: "THE_DATA"},
+                },
+            });
+            const result = cache.getEntry(fakeHandler, "options");
+
+            // Assert
+            expect(result).toStrictEqual({data: "THE_DATA"});
+        });
+
+        it("should throw if the cache is already intialized", () => {
+            // Arrange
+            const cache = new ResponseCache({
+                MY_HANDLER: {
+                    MY_KEY: {data: "THE_DATA"},
+                },
+            });
+
+            // Act
+            const underTest = () =>
+                cache.initialize({
+                    MY_HANDLER: {
+                        MY_OTHER_KEY: {data: "MORE_DATA"},
+                    },
+                });
+
+            // Assert
+            expect(underTest).toThrowErrorMatchingInlineSnapshot(
+                `"Cannot initialize data response cache more than once"`,
+            );
+        });
+
+        it("should deep clone the cached data", () => {
+            // Arrange
+            const cache = new ResponseCache();
+            const fakeHandler: IRequestHandler<string, string> = {
+                getKey: () => "MY_KEY",
+                type: "MY_HANDLER",
+                invalidateCache: () => false,
+                fulfillRequest: jest.fn(),
+            };
+            const sourceData = {
+                MY_HANDLER: {
+                    MY_KEY: {data: "THE_DATA"},
+                },
+            };
+
+            // Act
+            cache.initialize(sourceData);
+            // Try to mutate the cache.
+            sourceData["MY_HANDLER"]["MY_KEY"] = {data: "SOME_NEW_DATA"};
```

Comment:
I guess node isn't in strict mode.

Reply:
Not sure I understand. This is modifying this object:

```
const sourceData = {
                MY_HANDLER: {
                    MY_KEY: {data: "THE_DATA"},
                },
            };
```

Not the actual cache. It's showing that we're are indeed cloning rather than using it directly.

Would strict mode complain about that?

Reply:
For some reason I thought that `sourceData` was frozen.  Maybe it isn't.